### PR TITLE
CXP-1289: List JSON Schema targets opening the right panel on click

### PR DESCRIPTION
**Pull request by willy-ahva**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/catalogs/front/src/components/CatalogEdit/components/Edit.tsx [hunk 1]
```diff
@@ -13,13 +13,26 @@ import {getTabsValidationStatus} from '../utils/getTabsValidationStatus';
...
 
 type Props = {
     id: string;
     values: CatalogFormValues;
     errors: CatalogFormErrors;
 };
 
+const ProductMappingWrapper: FC<
+    PropsWithChildren<{
+        id: string;
```
Comment (masacc) [inline components/catalogs/front/src/components/CatalogEdit/components/Edit.tsx:27]:
```suggestion
        catalogId: string;
```

File: components/catalogs/front/src/components/CatalogEdit/hooks/useCatalog.ts [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
+import {ProductMapping} from '../../ProductMapping/models/ProductMapping';
 
 type Data = {
     id: string;
     name: string;
     enabled: boolean;
+    owner_username: string;
     product_selection_criteria: AnyCriterionState[];
     product_value_filters: ProductValueFiltersValues;
+    // TODO remove it
```
Comment (masacc) [inline components/catalogs/front/src/components/CatalogEdit/hooks/useCatalog.ts:13]: why ?

File: components/catalogs/front/src/components/CatalogEdit/reducers/CatalogFormReducer.unit.ts [hunk 3]
```diff
@@ -91,6 +99,48 @@ const tests: {state: CatalogFormValues; action: CatalogFormAction; result: Catal
                 locales: ['en_US', 'fr_FR'],
                 currencies: ['USD', 'EUR'],
             },
+            product_mapping: {},
+        },
+    },
+
```
Comment (masacc) [inline components/catalogs/front/src/components/CatalogEdit/reducers/CatalogFormReducer.unit.ts:105]: rm blank line

File: components/catalogs/back/src/Infrastructure/Persistence/Attribute/GetAttributeLabelsQuery.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+use Akeneo\Catalogs\Application\Persistence\Attribute\GetAttributeLabelsQueryInterface;
+use Akeneo\Pim\Structure\Component\Model\AttributeInterface;
+use Akeneo\Tool\Component\StorageUtils\Repository\SearchableRepositoryInterface;
+
+/**
+ * @author    Willy Mesnage <willy.mesnage@akeneo.com>
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class GetAttributeLabelsQuery implements GetAttributeLabelsQueryInterface
```
Comment (tseho) [inline components/catalogs/back/src/Infrastructure/Persistence/Attribute/GetAttributeLabelsQuery.php:16]: class to remove

File: components/catalogs/back/src/Application/Persistence/Attribute/GetAttributeLabelsQueryInterface.php [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+declare(strict_types=1);
+
+namespace Akeneo\Catalogs\Application\Persistence\Attribute;
+
+/**
+ * @author    Willy Mesnage <willy.mesnage@akeneo.com>
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface GetAttributeLabelsQueryInterface
```
Comment (tseho) [inline components/catalogs/back/src/Application/Persistence/Attribute/GetAttributeLabelsQueryInterface.php:12]: class to remove

File: components/catalogs/back/src/Infrastructure/Symfony/Resources/config/services.yml [hunk 6]
```diff
@@ -80,6 +80,10 @@ services:
         arguments:
             - '@pim_enrich.repository.attribute.search'
 
+    Akeneo\Catalogs\Infrastructure\Persistence\Attribute\GetAttributeLabelsQuery:
+        arguments:
+            - '@pim_enrich.repository.attribute.search'
+
```
Comment (tseho) [inline components/catalogs/back/src/Infrastructure/Symfony/Resources/config/services.yml:86]:
```suggestion
```

File: components/catalogs/front/src/components/ProductMapping/models/ProductMappingSchema.d.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    description: string;
+    type: string;
+    properties: {
+        [target: string]: {
+            title?: string;
+            description?: string;
+            type: string;
+        };
+    };
+};
```
Comment (masacc) [inline components/catalogs/front/src/components/ProductMapping/models/ProductMappingSchema.d.ts:15]: I think you only need that for now:
```suggestion
export type ProductMappingSchema = {
    properties: {
        [target: string]: {
            title?: string;
        };
    };
};
```

File: components/catalogs/front/src/hooks/useAttribute.ts [hunk 8]
```diff
@@ -1,5 +1,5 @@
 import {useQuery} from 'react-query';
-import {Attribute} from '../models/Attribute';
+import {Attribute} from '../components/ProductSelection/models/Attribute';
```
Comment (masacc) [inline components/catalogs/front/src/hooks/useAttribute.ts:2]: you should move this model at the same level as the hook using it

File: components/catalogs/front/src/components/CatalogEdit/hooks/useCatalog.ts (hunk 2 above)
Comment (guillaumefer-akeneo) [reply components/catalogs/front/src/components/CatalogEdit/hooks/useCatalog.ts]: it's an old comment for a removed hook. I remove it.

**Review by willy-ahva — COMMENTED** [review]
Can not approve my own PR but, I do!

**Review by masacc — APPROVED** [review]

**Review by RiffVentura — APPROVED** [review]

**Review by eoliveauakeneo — APPROVED** [review]

**Pull request merged**